lib/ddtrace/span/duration.rb

```diff
@@ -0,0 +1,70 @@
+require 'ddtrace/utils/time'
+
+module Datadog
+  class Span
+    # Stateful class used to track and calculate the duration of a Span.
+    class Duration
```

Comment:
Do we think this will be used elsewhere? Maybe a little overkill to encapsulate in a class?

Reply:
I tend to agree, I am probably going to refactor to be part of the `Span` class, just trying to get all the tests to pass with the current implementation, there are one or two corner cases still i believe.